### PR TITLE
en, zh: optimize upgrade-a-tidb-cluster

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added, or deleted? (Required)

Optimize upgrade-a-tidb-cluster.md:
- Add a section to introduce rolling update and its impact on online applications.
- Reorganize the structure:
    -  `force-upgrade` is a troubleshooting guide for rolling update.
    - `修改 TiDB 集群配置` has nothing to do with rolling update, and should be a standalone doc under `Manage > Maintain`.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/modify-tidb-configuration.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    当所有 Pod 都重建完毕进入 `Running` 状态后，配置修改完成。
+
+## 修改 PD 组件配置
+
+在 PD 首次启动成功后，PD 的部分配置项会持久化到 etcd 中，且后续将以 etcd 中的配置为准。因此，在 PD 首次启动后，这些配置项将无法再通过 TidbCluster CR 来进行修改。
+
+PD 中[支持在线修改的配置项](https://docs.pingcap.com/zh/tidb/stable/dynamic-config#在线修改-pd-配置)里，除 `log.level` 外，其他配置项在 PD 首次启动之后均不再支持通过 TidbCluster CR 进行修改。
+
+对于部署在 Kubernetes 中的 TiDB 集群，如需修改 PD 配置参数，需要使用 SQL、pd-ctl 或 PD server API 来动态进行修改。//TODO：提供 SQL、pd-ctl 或 PD server API 的使用方式或文档链接
```
Comment (ran-huang) [inline zh/modify-tidb-configuration.md:42]: @DanielZhangQD 看下 TODO

File: zh/upgrade-a-tidb-cluster.md [hunk 2]
```diff
@@ -47,59 +55,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-a-tidb-cluster/']
...
     {{< copyable "shell-regular" >}}
 
-    ```shell
-    kubectl edit tc ${cluster_name} -n ${namespace}
+    ```bash
+    kubectl annotate --overwrite tc ${cluster_name} -n ${namespace} tidb.pingcap.com/force-upgrade=true
     ```
 
-2. 查看配置修改后的更新进度：
+2. 修改 PD 相关配置，确保 PD 进入正常状态。// TODO：这里是否要 refer 到[修改 TiDB 集群配置](zh/modify-tidb-configuration.md) 这个文档？
```
Comment (ran-huang) [inline zh/upgrade-a-tidb-cluster.md:76]: @DanielZhangQD TODO 👀

File: zh/modify-tidb-configuration.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+title: 修改 TiDB 集群配置
+summary: 了解如何修改部署在 Kubernetes 上的 TiDB 的集群配置。
+---
+
+# 修改 TiDB 集群配置
+
+TiDB 集群自身支持通过 SQL 对各组件进行[在线配置变更](https://docs.pingcap.com/zh/tidb/stable/dynamic-config)，无需重启集群组件。但是，对于部署在 Kubernetes 中的 TiDB 集群，部分组件在升级后，配置项会被 TidbCluster CR 中的配置项覆盖，导致在线变更的配置失效。
```
Comment (DanielZhangQD) [inline zh/modify-tidb-configuration.md:8]:
```suggestion
TiDB 集群自身支持通过 SQL 对各组件进行[在线配置变更](https://docs.pingcap.com/zh/tidb/stable/dynamic-config)，无需重启集群组件。但是，对于部署在 Kubernetes 中的 TiDB 集群，部分组件在升级或者重启后，配置项会被 TidbCluster CR 中的配置项覆盖，导致在线变更的配置失效。
```

File: zh/modify-tidb-configuration.md (hunk 1 above)
Comment (DanielZhangQD) [reply zh/modify-tidb-configuration.md:42]: pd-ctl: https://docs.pingcap.com/zh/tidb/dev/pd-control/#config-show--set-option-value--placement-rules
sql: https://docs.pingcap.com/zh/tidb/stable/dynamic-config/#%E5%9C%A8%E7%BA%BF%E4%BF%AE%E6%94%B9-pd-%E9%85%8D%E7%BD%AE
pd api 好像没有文档介绍

File: zh/upgrade-a-tidb-cluster.md [hunk 4]
```diff
@@ -1,40 +1,48 @@
...
+## 功能介绍
 
-滚动更新可以用于升级 TiDB 版本，也可以用于更新集群配置。
+Kubernetes 提供[滚动更新功能](https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-intro/)，在不影响应用可用性的前提下执行更新。
 
-## 通过 TidbCluster CR 升级
+使用滚动更新时，TiDB Operator 会按 PD、TiKV、TiDB 的顺序，串行地删除旧版本的 Pod，并创建新版本的 Pod。当新版本的 Pod 正常运行后，再处理下一个 Pod。
 
-如果 TiDB 集群是直接通过 TidbCluster CR 部署的或者通过 Helm 方式部署的然后又切换到了 TidbCluster CR 管理，可以通过下面步骤升级 TiDB 集群。
+滚动更新中，TiDB Operator 会自动处理 PD 和 TiKV 的 Leader 迁移与 TiDB 的 DDL Owner 迁移。因此，在多节点的部署拓扑下（最小环境：PD \* 3、TiKV \* 3、TiDB \* 2），滚动更新 TiKV、PD 不会影响业务正常运行。对于有连接重试功能的客户端，滚动更新 TiDB 同样不会影响业务。
```
Comment (DanielZhangQD) [inline zh/upgrade-a-tidb-cluster.md:19]:
```suggestion
滚动更新中，TiDB Operator 会自动处理 PD 和 TiKV 的 Leader 迁移。因此，在多节点的部署拓扑下（最小环境：PD \* 3、TiKV \* 3、TiDB \* 2），滚动更新 TiKV、PD 不会影响业务正常运行。对于有连接重试功能的客户端，滚动更新 TiDB 同样不会影响业务。
```

File: zh/upgrade-a-tidb-cluster.md [hunk 5]
```diff
@@ -47,59 +55,32 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-a-tidb-cluster/']
 
 > **注意：**
 >
-> 如果需要升级到企业版，需要将 db.yaml 中 `spec.<tidb/pd/tikv/tiflash/ticdc/pump>.baseImage` 配置为企业版镜像，格式为 `pingcap/<tidb/pd/tikv/tiflash/ticdc/tidb-binlog>-enterprise`。
+> 如果需要升级到企业版，需要将 `db.yaml` 中 `spec.<tidb/pd/tikv/tiflash/ticdc/pump>.baseImage` 配置为企业版镜像，格式为 `pingcap/<tidb/pd/tikv/tiflash/ticdc/tidb-binlog>-enterprise`。
```
Comment (DanielZhangQD) [inline zh/upgrade-a-tidb-cluster.md:58]:
```suggestion
> 如果需要升级到企业版，需要将 `spec.<tidb/pd/tikv/tiflash/ticdc/pump>.baseImage` 配置为企业版镜像，格式为 `pingcap/<tidb/pd/tikv/tiflash/ticdc/tidb-binlog>-enterprise`。
```

File: zh/upgrade-a-tidb-cluster.md (hunk 2 above)
Comment (DanielZhangQD) [reply zh/upgrade-a-tidb-cluster.md:76]: 不用吧

File: zh/modify-tidb-configuration.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
+title: 修改 TiDB 集群配置
+summary: 了解如何修改部署在 Kubernetes 上的 TiDB 的集群配置。
+---
+
+# 修改 TiDB 集群配置
+
+TiDB 集群自身支持通过 SQL 对各组件进行[在线配置变更](https://docs.pingcap.com/zh/tidb/stable/dynamic-config)，无需重启集群组件。但是，对于部署在 Kubernetes 中的 TiDB 集群，部分组件在升级或者重启后，配置项会被 TidbCluster CR 中的配置项覆盖，导致在线变更的配置失效。
```
Comment (DanielZhangQD) [inline zh/modify-tidb-configuration.md:8]: 看文档里面只有 PD, TiDB, TiKV, 这里要说清楚吗？

File: zh/modify-tidb-configuration.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+本文介绍如何修改部署在 Kubernetes 上的 TiDB 的集群配置。由于 PD 的特殊性，需要分别对 PD 和其他组件进行配置。
+
+## 修改 TiDB/TiKV 等组件配置
+
+除 PD 外的组件（TiDB、TiKV、TiFlash、TiCDC 等），升级后配置项会被 TidbCluster CR 中的配置项覆盖，导致在线变更的配置失效。因此，对于这些组件，你需要在 TidbCluster CR 中直接修改配置项。
```
Comment (DanielZhangQD) [inline zh/modify-tidb-configuration.md:14]:
```suggestion
对于 TiDB 和 TiKV，如果通过 SQL 进行[在线配置变更](https://docs.pingcap.com/zh/tidb/stable/dynamic-config)，在升级或者重启后，配置项会被 TidbCluster CR 中的配置项覆盖，导致在线变更的配置失效。因此，如果需要持久化修改配置，你需要在 TidbCluster CR 中直接修改配置项。
对于 TiFlash、TiCDC 和 Pump，目前只能通过在 TidbCluster CR 中修改配置项。
```

File: zh/modify-tidb-configuration.md (hunk 6 above)
Comment (ran-huang) [reply zh/modify-tidb-configuration.md]: 加了 `TiDB 集群自身支持通过 SQL 对 TiDB、TiKV、PD 等组件进行[在线配置变更]`

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by lilin90 — APPROVED** [review]
LGTM

**Comment by ran-huang** [comment]
@DanielZhangQD @lilin90 Please review the English version. Thanks!

File: en/modify-tidb-configuration.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
+---
+title: Modify TiDB cluster configuration
+summary: Learn how to modify the configuration of TiDB clusters deployed in Kubernetes.
+---
+
+# Modify TiDB cluster configuration
+
+For a TiDB cluster, you can [update the configuration of components](https://docs.pingcap.com/tidb/stable/dynamic-config/) online using SQL statements, including TiDB, TiKV, and PD, without restarting the cluster components. However, for TiDB clusters deployed in Kubernetes, after you upgrade or restart the cluster, the configurations of some components might be overwritten by the `TidbCluster` CR. This leads to invalid online configuration update.
```
Comment (DanielZhangQD) [inline en/modify-tidb-configuration.md:8]:
```suggestion
For a TiDB cluster, you can [update the configuration of components](https://docs.pingcap.com/tidb/stable/dynamic-config/) online using SQL statements, including TiDB, TiKV, and PD, without restarting the cluster components. However, for TiDB clusters deployed in Kubernetes, after you upgrade or restart the cluster, the configurations updated using SQL statements will be overwritten by those in the `TidbCluster` CR. This leads to the online configuration update being invalid.
```

File: en/modify-tidb-configuration.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+This document describes how to modify the configuration of TiDB clusters deployed in Kubernetes. Due to the special nature of PD, you need to separately modify the configuration of PD and other components.
+
+## Modify configuration for TiDB, TiKV, and other components
+
+For TiDB and TiKV, if you [dynamically modify their configuration](https://docs.pingcap.com/tidb/stable/dynamic-config/) using SQL statements, after you upgrade or restart the cluster, the configurations of some components might be overwritten by the `TidbCluster` CR. This leads to invalid online configuration update. Therefore, to persist the configuration, you must directly modify their configurations in the `TidbCluster` CR.
```
Comment (DanielZhangQD) [inline en/modify-tidb-configuration.md:14]:
```suggestion
For TiDB and TiKV, if you [dynamically modify their configuration](https://docs.pingcap.com/tidb/stable/dynamic-config/) using SQL statements, after you upgrade or restart the cluster, the configurations will be overwritten by those in the `TidbCluster` CR. This leads to the online configuration update being invalid. Therefore, to persist the configuration, you must directly modify their configurations in the `TidbCluster` CR.
```

File: en/modify-tidb-configuration.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+    ```shell
+    watch kubectl -n ${namespace} get pod -o wide
+    ```
+
+    After all the Pods are recreated and are in the `Running` state, the configuration is successfully modified.
+
+## Modify PD configuration
+
+After PD is started for the first time, some PD configuration items are persisted in etcd. The persisted configuration in etcd takes precedence over the configuration in PD. Therefore, after the first start, you cannot modify some PD configuration by using the `TidbCluster` CR.
```
Comment (DanielZhangQD) [inline en/modify-tidb-configuration.md:40]:
```suggestion
After PD is started for the first time, some PD configuration items are persisted in etcd. The persisted configuration in etcd takes precedence over the configuration file in PD. Therefore, after the first start, you cannot modify some PD configuration by using the `TidbCluster` CR.
```

File: en/modify-tidb-configuration.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+Among all the PD configuration items listed in [Modify PD configuration online](https://docs.pingcap.com/tidb/stable/dynamic-config/#modify-pd-configuration-online), after the first start, only `log.level` can be modified by using the `TidbCluster` CR. Other configurations cannot be modified by using CR.
+
+For TiDB clusters deployed in Kubernetes, if you need to modify the PD configuration, you  can dynamically modify the configuration using [SQL statements](https://docs.pingcap.com/tidb/stable/dynamic-config/#modify-pd-configuration-online), [pd-ctl](https://docs.pingcap.com/tidb/dev/pd-control/#config-show--set-option-value--placement-rules), or PD server API.
```
Comment (DanielZhangQD) [inline en/modify-tidb-configuration.md:44]:
```suggestion
For TiDB clusters deployed in Kubernetes, if you need to modify the PD configuration, you can dynamically modify the configuration using [SQL statements](https://docs.pingcap.com/tidb/stable/dynamic-config/#modify-pd-configuration-online), [pd-ctl](https://docs.pingcap.com/tidb/dev/pd-control/#config-show--set-option-value--placement-rules), or PD server API.
```

File: en/modify-tidb-configuration.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
+---
+title: Modify TiDB cluster configuration
```
Comment (lilin90) [inline en/modify-tidb-configuration.md:2]:
```suggestion
title: Modify TiDB Cluster Configuration
```

File: en/modify-tidb-configuration.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+summary: Learn how to modify the configuration of TiDB clusters deployed in Kubernetes.
+---
+
+# Modify TiDB cluster configuration
```
Comment (lilin90) [inline en/modify-tidb-configuration.md:6]:
```suggestion
# Modify TiDB Cluster Configuration
```

File: en/modify-tidb-configuration.md (hunk 8 above)
Comment (lilin90) [reply en/modify-tidb-configuration.md:8]: I see that we use both `online` and `dynamically` in documentation. How about trying to keep the word usage consistent? Which one is better in your opinion?

Comment (ran-huang) [reply en/modify-tidb-configuration.md:8]: I prefer `online`. TiDB documentation uses `online` almost across the [whole doc](https://docs.pingcap.com/tidb/stable/dynamic-config) except in the URL slug.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**